planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp

```diff
@@ -212,6 +223,29 @@ TEST_F(TestNormalLaneChange, testGetPathWhenInvalid)
   ASSERT_FALSE(lc_status.is_valid_path);
 }
 
+TEST_F(TestNormalLaneChange, testFilteredObjects)
+{
+  constexpr auto is_approved = true;
+  ego_pose_ = autoware::test_utils::createPose(1.0, 1.75, 0.0, 0.0, 0.0, 0.0);
+  planner_data_->self_odometry = set_odometry(ego_pose_);
+  set_previous_approved_path();
+
+  normal_lane_change_->update_lanes(!is_approved);
+  normal_lane_change_->update_filtered_objects();
+
+  const auto & filtered_objects = get_filtered_objects();
+
+  // Note: There's 1 stopping object in current lanes, however, it was filtered out.
+  const auto filtered_size =
+    filtered_objects.current_lane.size() + filtered_objects.target_lane_leading.size() +
+    filtered_objects.target_lane_trailing.size() + filtered_objects.others.size();
+  ASSERT_EQ(filtered_size, planner_data_->dynamic_object->objects.size());
+  ASSERT_EQ(filtered_objects.current_lane.size(), 0);
+  ASSERT_EQ(filtered_objects.target_lane_leading.size(), 2);
+  ASSERT_EQ(filtered_objects.target_lane_trailing.size(), 0);
+  ASSERT_EQ(filtered_objects.others.size(), 2);
```

Comment:
`ASSERT_EQ` will terminate the test on first failure, maybe it makes more sense to use `EXPECT_EQ` so you can ge the result of all even if once fails.

Reply:
Sure, changed it here https://github.com/autowarefoundation/autoware.universe/pull/9256/commits/6a83976df7f58e93caeecf48348651317eb7de52